lib/heap_sort.rb

```diff
@@ -1,8 +1,19 @@
 
 
 # This method uses a heap to sort an array.
-# Time Complexity:  ?
-# Space Complexity: ?
-def heap_sort(list)
-  raise NotImplementedError, "Method not implemented yet..."
+# Time Complexity:  O(1)
+# Space Complexity: O(n)
+def heapsort(array)
```

Comment:
Time complexity O(1)?  How?  Heapsort is actually O(n log n).

This method does however work.